src/coreclr/jit/runtimelookup.cpp

```diff
@@ -238,14 +244,7 @@ PhaseStatus Compiler::fgExpandRuntimeLookups()
                 BasicBlock* fallbackBb = fgNewBBafter(BBJ_NONE, nullcheckBb, true);
                 fallbackBb->bbFlags |= BBF_INTERNAL;
 
-                GenTree* signatureArg =
-                    gtNewIconEmbHndNode(signature, nullptr, GTF_ICON_GLOBAL_PTR,
-                                        (void*)sigTree->gtEffectiveVal()->AsIntCon()->gtCompileTimeHandle);
-                fgUpdateConstTreeValueNumber(signatureArg);
-                GenTreeCall* fallbackCall =
-                    gtNewHelperCallNode(runtimeLookup.helper, TYP_I_IMPL, gtClone(ctxTree), signatureArg);
-                gtSetEvalOrder(fallbackCall);
-                fgMorphCall(fallbackCall);
+                GenTreeCall* fallbackCall = gtCloneExpr(call)->AsCall();
                 assert(!fallbackCall->IsExpRuntimeLookup());
                 assert(ctxTree->OperIs(GT_LCL_VAR));
                 Statement* asgFallbackStmt = fgNewStmtFromTree(gtNewAssignNode(gtClone(rtLookupLcl), fallbackCall));
```

Comment:
I think now we can just use the original call here, and below where you do `call->BashToLclVar` we can just update `callUse` to be a use of this assigned local.

Reply:
Done, thanks!